webportal/src/app/market_list/components/side_bar.jsx

```diff
@@ -1,59 +1,68 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
-import React, { useContext, useState } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { getTheme, Stack, Text, SearchBox } from 'office-ui-fabric-react';
-import queryString from 'query-string';
-import { isNil } from 'lodash';
-import TypeFilter from './type_filter';
+import { isEmpty } from 'lodash';
+import FilterItem from './type_filter';
 import Context from 'App/context';
 import { TYPE_ENUM } from 'App/utils/constants';
+import PropTypes from 'prop-types';
 
-const SideBar = () => {
-  const [type, setType] = useState(null);
-  const [keyword, setKeyword] = useState();
+const SideBar = props => {
+  const { categories } = props;
+  const [currentType, setCurrentType] = useState(null);
+  const [currentAuthor, setCurrentAuthor] = useState(null);
+  const [currentCategory, setCurrentCategory] = useState(null);
   const { spacing } = getTheme();
-  const { history, user } = useContext(Context);
+  const { history, user, routeProps } = useContext(Context);
 
-  const getQueryString = (type, keyword) => {
-    let qs = '';
-    if (isNil(keyword) || keyword === '') {
-      if (isNil(type)) {
-        qs = '';
-      } else if (type === 'my') {
-        qs = queryString.stringify({ author: user });
-      } else {
-        qs = queryString.stringify({ type });
-      }
-    } else {
-      if (isNil(type)) {
-        qs = queryString.stringify({ keyword });
-      } else if (type === 'my') {
-        qs = queryString.stringify({ author: user, keyword });
-      } else {
-        qs = queryString.stringify({ type, keyword });
-      }
-    }
-    return qs;
-  };
+  useEffect(() => {
+    const searchParams = new URLSearchParams(routeProps.location.search);
+    setCurrentType(searchParams.get('type'));
+    setCurrentAuthor(searchParams.get('author'));
+    setCurrentCategory(searchParams.get('category'));
+  }, []);
 
-  const changeType = newType => {
+  const changeUrl = (section, value) => {
     return () => {
-      if (type === newType) {
-        setType(null);
-        const qs = getQueryString(null, keyword);
-        history.push(`/?${qs}`);
-      } else {
-        setType(newType);
-        const qs = getQueryString(newType, keyword);
-        history.push(`/?${qs}`);
+      const searchParams = new URLSearchParams(routeProps.location.search);
+      if (section === 'keyword') {
+        if (value === '') {
```

Comment:
>value [](http://example.com/codeflow?start=12&length=5)

do we need to check if the vale is null or not?

Reply:
and for searchParams, deleting a non-existing item is non exceptionable. right?

Reply:
> > value
> 
> do we need to check if the vale is null or not?

In the search component, value could not be null or not if called this method as 'keyword' section.

Reply:
> and for searchParams, deleting a non-existing item is non exceptionable. right?

I have tested that the delete function allows delete non-existing item, nothing happens and no exception will be thrown